components/channel_header_dropdown/menu_items/close_direct_channel/close_direct_channel.js

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {localizeMessage} from 'utils/utils';
+import {Constants} from 'utils/constants';
+import {browserHistory} from 'utils/browser_history';
+import Menu from 'components/widgets/menu/menu';
+
+export default class CloseDirectChannel extends React.PureComponent {
+    static propTypes = {
+
+        /**
+         * Object with info about currentUser
+         */
+        currentUser: PropTypes.object.isRequired,
+
+        /**
+         * Object with info about currentTeam
+         */
+        currentTeam: PropTypes.object.isRequired,
+
+        /**
+         * String with info about redirect
+         */
+        redirectChannel: PropTypes.string.isRequired,
+
+        /**
+         * Object with info about channel
+         */
+        channel: PropTypes.object.isRequired,
+
+        /**
+         * Use for test selector
+         */
+        id: PropTypes.string,
+
+        /**
+         * Object with action creators
+         */
+        actions: PropTypes.shape({
+
+            /**
+             * Action creator to update user preferences
+             */
+            savePreferences: PropTypes.func.isRequired,
+        }).isRequired,
+    };
+
+    isLeaving = false;
```

Comment:
It looks like you meant to use it to prevent double clicking `handleClose`, but if it's not needed for that, would you mind removing it?